scripts/mods/msu/persistent_data_manager.nut

```diff
@@ -0,0 +1,99 @@
+this.persistent_data_manager <- {
+	m = {
+		ModConfigPath = "mod_config/"
+		Mods = {}
+	},
+	
+	function create()
+	{
+		this.importModConfigFiles();
+	}
+
+	function importModConfigFiles()
+	{
+		local persistentDirectory = this.IO.enumerateFiles(this.m.ModConfigPath);
+		if (persistentDirectory == null)
+		{
+			return;
+		}
+		foreach (file in persistentDirectory)
+		{
+			local components = split(file, "/");
+			local modID = components[1];
+			local settingsType = components[2];
+			::MSU.Mod.Debug.printWarning(format("Checking file, potential modID: '%s' and settingstype '%s'.", modID, settingsType), "persistence");
+			this.setMod(modID);
+			if (!(settingsType in this.getMods()[modID]))
+			{
+				this.getMods()[modID][settingsType] <- file;
+			}
+		}
+	}
+	
+
+	function getMods()
+	{
+		return this.m.Mods;
+	}
+
+	function hasMod( _modID )
+	{
+		return (_modID in this.getMods());
+	}
+
+	function getMod(_modID)
+	{
+		if (!this.hasMod(_modID))
+		{
+			throw ("Mod " + _modID + " not found in mods!");
+		}
+		return this.getMods()[_modID];
+	}
+
+	function loadSettingForMod( _modID, _settingID )
+	{
+		::MSU.Mod.Debug.printWarning(format("Loading setting '%s' for mod '%s'.", _settingID, _modID), "persistence");
+		if (_settingID in this.getMod(_modID))
+		{
+			::include(this.getMod(_modID)[_settingID]);
+			return true;
+		}
+		return false;
+
+	}
+
+	function loadSettingForEveryMod( _settingID )
+	{
+		foreach (modID, modValue in this.getMods())
+		{
+			this.loadSettingForMod(modID, _settingID);
+		}
+	}
+
+	function loadAllSettingsForMod( _modID, _settingID )
+	{
+		if (!this.hasMod(_modID))
+		{
+			throw ("Mod " + _modID + " not found in mods!");
+		}
+
+		foreach (setting in this.getMod(_modID))
+		{
+			this.loadSettingForMod(_modID, setting);
+		}
+	}
+
+	function setMod( _modID, _reset = false )
+	{
+		if (this.hasMod(_modID) && _reset == false)
+		{
+			return;
+		}
+		this.getMods()[_modID] <- {};
+	}
+
+	function writeToLog(_settingID, _modID, _value)
+	{
+		::logInfo(format("PARSEME;%s;%s;%s", _settingID.tostring(), _modID.tostring(), _value.tostring()));
```

Comment:
Now that I noticed it it rly bothers me that this is in order of settingID, modID, value lol

Reply:
Yeah modID, settingID, value is better.

Reply:
Based on the internal workings of bbparser this makes more sense. Also, I should change `PARSEME` to `BBPARSER` all-around.

Reply:
Strongly agree with changing `PARSEME` to something else. I'd suggest `BBPARSER-PARSEME` because we may wanna output logs from `BBPARSER` which may not need to be parsed?